resmem.c

```diff
@@ -19,14 +19,16 @@
 #ifndef _RESOURCE_H
 #include "resource.h"
 #endif
-#include <malloc.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include "resmem.h"
 #include "resource_impl.h"
 #include <errno.h>
+#include <libgen.h>
+
+#define startswith(str, buf) strncmp(str, buf, sizeof(str) - 1)
```

Comment:
I think it's important to get an intuitive return value here.  Why not something like

```
#define startswith(str, buf) (strncmp(str, buf, sizeof(str) - 1) == 0)
```
so the caller can just check whether startswith() returns true?

Reply:
I was leaving the interpretation or return value of macro on the user. However all places I have used it in, use it for a true/false return. So I can make this change.